Simplified/Catalog/NYPLCatalogUngroupedFeed.m

```diff
@@ -27,31 +27,55 @@ @interface NYPLCatalogUngroupedFeed ()
 // attempt to fetch more books will be made.
 static NSUInteger const preloadThreshold = 100;
 
+// The current number of attempts of books fetching until we receive supported books,
+// and the maximum attempts allowed.
+static NSInteger fetchSupportedBooksRetryCount = 0;
+static NSInteger const fetchSupportedBooksRetryThreshold = 15;
+
 @implementation NYPLCatalogUngroupedFeed
 
-+ (void)withURL:(NSURL *)URL
-handler:(void (^)(NYPLCatalogUngroupedFeed *category))handler
-{
++ (void)fetchSupportedBooksWithURL:(NSURL *)URL
+                          provider:(Class <NYPLOPDSFeedProvider>)provider
+                           handler:(void (^)(NYPLCatalogUngroupedFeed *feed))handler {
```

Comment:
is this handler a completion handler, semantically?

Reply:
Yes, I will rename it for clarity